code/gateway/dataaccess.q

```diff
@@ -171,27 +173,45 @@ partdict:{[input]
 adjustqueries:{[options;part]
     // if only one process then no need to adjust
     if[2>count p:options`procs;:options];
-    // get the date casting where relevant
-    st:$[a:-14h~tp:type start:options`starttime;start;`date$start];
-    et:$[a;options`endtime;`date$options`endtime];
-    // get the dates that are required by each process
-    dates:group key[part]where each{within[y;]each value x}[part]'[l:st+til 1+et-st];
-    dates:l{(min x;max x)}'[dates];
-    // if start/end time not a date, then adjust dates parameter for the
-    // correct types
-    if[not a;
-        // converts dates dictionary to timestamps/datetimes
-        dates:$[-15h~tp;{"z"$x};::]{(0D+x 0;x[1]+1D-1)}'[dates];
+    // get the tablename
+    tabname:options[`tablename];
+    // remove duplicate servertypes from the gw.servers
+    servers:select from .gw.servers where i=(first;i)fby servertype;
+    // extract the procs which have the table defined
+    servers:select from servers where {[x;tabname]tabname in @[x;`tables]}[;tabname] each attributes;
+    // create a dictionary of the attributes against servertypes
+    procdict:(exec servertype from servers)!(exec attributes from servers)@'(key each exec attributes from servers)[;0];
```

Comment:
if I'm reading this right, this assumes that the partition type will always be the first key in the attributes dict, is that a safe assumption? generally I don't like enforcing a rigid ordering for dictionaries, one of the big benefits of dictionaries over lists is that we _don't_ have to have things in a particular order as we can lookup by key instead of index.

Perhaps instead we could add a `partitiontype` key to the attributes dict, and use that?

Reply:
I think you are right, this part is existing logic taken from the .dataaccess.partdict[] function. Can certainly look at changing it but presumably would need to change that existing funciton as well then?

Reply:
I think that's OK, I don't think it should be a massive change hopefully

Reply:
formatting is a bit messed up there https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax

I _think_ what you're saying makes sense but hard to see exactly what you're saying

Reply:
sorry hopefully that has fixed it

Reply:
yep looks good - I think you can probably skip the lambda within something like

```q
procdict:(exec servertype from servers)!(exec attributes from servers)[;`partition]
```